### PR TITLE
Fix segfault in error note

### PR DESCRIPTION
**Pull request by tomc1998**
The following code produces a segfault in current master:

```
fn init() !u32 {
    return 0;
}

const Foo = struct {
    data: u32 = init(),
};

pub fn main() void {
    const state: Foo = .{};
    _ = state;
}
```

This is caused by default assigning `data` with an error union value when it doesn't expect one - it segfaults when it tries to add a compiler note. I think because it's trying to evaluate the `init()` body as a constant expression but the compiler note expects it to be inside a function (?)

This change just avoids adding the compiler note in this case. The error still reads fine:

```
./zig-bounded-array-repro.zig:6:21: error: cannot convert error union to payload type. consider using `try`, `catch`, or `if`. expected type 'u32', found '@typeInfo(@typeInfo(@TypeOf(init)).Fn.return_type.?).ErrorUnion.error_set!u32'
    data: u32 = init(),
```

### CODE REVIEW

File: src/stage1/ir.cpp [hunk 1]
```diff
@@ -15500,7 +15500,7 @@ static Stage1AirInst *ir_analyze_struct_field_ptr(IrAnalyze *ira, Scope *scope,
                 is_const, is_volatile, PtrLenSingle, field->align,
                 (uint32_t)(ptr_bit_offset + field->bit_offset_in_host),
                 (uint32_t)host_int_bytes_for_result_type, false);
-        
+
```
Comment (vincenzopalazzo) [inline src/stage1/ir.cpp:15503]: maybe this is an unnecessary change?

Comment (viriuwu) [reply src/stage1/ir.cpp:15503]: what's wrong with removing trailing whitespace? it's clearly a mistake from an editor not configured to remove them on save.

Comment (vincenzopalazzo) [reply src/stage1/ir.cpp:15503]: nothing wrong, I don't suggest a change, I made just a question, right? 

this can be a mistake fixed, or just a mistake added. I just asked to be sure!

Comment (tomc1998) [reply src/stage1/ir.cpp:15503]: Oh yeah, my emacs is setup to remove trailing whitespace, do we want this back in?

Comment (vincenzopalazzo) [reply src/stage1/ir.cpp:15503]: For me it is fine! I just point out this change!
